src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Policy/DefaultAuthorizationContext.cs

```diff
@@ -0,0 +1,61 @@
+﻿using CoreWCF.IdentityModel.Claims;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Text;
+
+namespace CoreWCF.IdentityModel.Policy
+{
+   internal class DefaultAuthorizationContext : AuthorizationContext
+    {
+        private static DefaultAuthorizationContext empty;
+        SecurityUniqueId id;
+        ReadOnlyCollection<ClaimSet> claimSets;
+        DateTime expirationTime;
+        IDictionary<string, object> properties;
+
+        public DefaultAuthorizationContext(DefaultEvaluationContext evaluationContext)
+        {
+            this.claimSets = evaluationContext.ClaimSets;
+            this.expirationTime = evaluationContext.ExpirationTime;
+            this.properties = evaluationContext.Properties;
+        }
+
+        public static DefaultAuthorizationContext Empty
+        {
+            get
+            {
+               if(empty == null)
+                    empty = new DefaultAuthorizationContext(new DefaultEvaluationContext());
```

Comment:
This code is still using a cached single DefaultAuthorizationContext. We need to either return an immutable context or not cache a single instance. As this code currently is written, a badly written authorization manager can leak claims between calls.